requirements.txt

```diff
@@ -1,7 +1,6 @@
 cosl
-ops>=2.0.0
-lightkube
-lightkube-models
+dacite
 jinja2
-jsonschema
-pydantic<2.0
+ops>=2.0.0
+pydantic~=2.8.2
+yarl
```

Comment:
This is for handling URLs operations.

Reply:
Do we really need this? It only provides syntactic sugar for URL operations, nothing the std lib cannot handle.
I think we can live without it and not bloat the project.

If I'm missing something and you we need to keep it, you could remove it from here and put it in `integration-requirements.txt`, I see it is only used in integration tests. Correct me if I'm wrong

Reply:
I think it's also used in the source code, not only in the tests. The stdlib cannot take care of the relative path as we want in the charm. (see [this](https://stackoverflow.com/questions/10893374/python-confusions-with-urljoin/10893427#10893427))